### PR TITLE
Add temporary logging on json metadata hash

### PR DESCRIPTION
**Pull request by bbaker6225**
## Description

Description of changes made

## Ticket Link

Link to the associated ticket
[Link title](https://openedx.atlassian.net/browse/ENT-XXXX)

## Post-review

Squash commits into discrete sets of changes

### CODE REVIEW

File: enterprise_catalog/apps/catalog/utils.py [hunk 1]
```diff
@@ -25,6 +25,18 @@ def get_content_filter_hash(content_filter):
...
 
 
+def get_metadata_hash(metadata):
+    """
+    This is a temporary duplicate of get_content_filter_hash for logging
+
+    This will likely be removed soon, but could also be updated to hash more of a content metadata
+    object as opposed to simply json_metadata
+    """
+    metadata_sorted_keys = json.dumps(metadata, sort_keys=True).encode()
```
Comment (iloveagent57) [inline enterprise_catalog/apps/catalog/utils.py]: what if the keys are valued by lists that are identical, but sorted differently?

Comment (iloveagent57) [reply enterprise_catalog/apps/catalog/utils.py]: I wonder if it's better (and any slower?) to just do a straight dictionary comparison?

Comment (iloveagent57) [reply enterprise_catalog/apps/catalog/utils.py]: ...or sort the JSON representation and then do a string comparison?
```python
>>> a = {'a': [1, 2, 3], 'b': [4, 5, 6]}
>>> d1 = {'a': [1, 2, 3], 'b': [4, 5, 6]}
>>> d2 = {'a': [3, 2, 1], 'b': [6, 5, 4]}
>>> d1 == d2
False
>>> import json
>>> sorted(json.dumps(d1))
[' ', ' ', ' ', ' ', ' ', ' ', ' ', '"', '"', '"', '"', ',', ',', ',', ',', ',', '1', '2', '3', '4', '5', '6', ':', ':', '[', '[', ']', ']', 'a', 'b', '{', '}']
>>> sorted(json.dumps(d1)) == sorted(json.dumps(d2))
True
```

File: enterprise_catalog/apps/catalog/models.py [hunk 2]
```diff
@@ -295,6 +296,20 @@ def associate_content_metadata_with_query(metadata, catalog_query):
...
+            old_metadata = None
+
+        if old_metadata:
+            old_json_metadata_hash = get_metadata_hash(old_metadata.json_metadata)
+            new_json_metadata_hash = get_metadata_hash(entry)
+            if old_json_metadata_hash != new_json_metadata_hash:
+                LOGGER.info(
+                    'JSON metadata hash for content key %s has changed',
+                    content_key,
+                )
```
Comment (iloveagent57) [inline enterprise_catalog/apps/catalog/models.py]: I think it'll be easier to count how many times the metadata hasn't changed.

**Review by adamstankiewicz — APPROVED** [review]
LGTM, but will defer to @iloveagent57 for the final 👍since you addressed his feedback :)

**Review by iloveagent57 — APPROVED** [review]
I have a feeling I'm overlooking something with the simplicity of the "sort the JSON strings" approach, but heck, let's try it.

**Pull request merged**